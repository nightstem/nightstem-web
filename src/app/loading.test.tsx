import { axe } from 'vitest-axe';
import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';

import LoadingPage from '@/app/loading';

describe(LoadingPage, () => {
  it('does not have accessibility violations', async () => {
    const { container } = render(<LoadingPage />);
    const result = await axe(container);

    expect(result).toHaveNoViolations();
  });

  it('matches the snapshot', () => {
    const { container } = render(<LoadingPage />);

    expect(container).toMatchSnapshot();
  });
});
