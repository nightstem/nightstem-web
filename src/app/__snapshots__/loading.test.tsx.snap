// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingPage > matches the snapshot 1`] = `
<div>
  <main
    class="h-full w-full"
  >
    <div
      aria-busy="true"
      class="grid h-full w-full place-items-center bg-background"
    >
      <div
        aria-live="polite"
        class="grid justify-items-center gap-4"
        role="status"
      >
        <div
          class="inline-flex"
        >
          <svg
            aria-label="Nightstem logo"
            class="text-primary-500 h-[4.488rem] w-[4.488rem] animate-pulse motion-reduce:animate-none shrink-0"
            fill="none"
            role="img"
            viewBox="0 0 500 500"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M346.04 135.875L500 245.427L161.185 495.252C157.629 497.874 151.828 497.923 148.228 495.362L0.786259 390.448L346.04 135.875Z"
              fill="currentColor"
            />
            <path
              d="M148.322 4.74771C151.878 2.12562 157.68 2.07657 161.279 4.63814L333.785 127.386L179.023 241.499L9.54552e-05 114.113L148.322 4.74771Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <p
          class="text-foreground/70"
        >
          Loading...
        </p>
      </div>
    </div>
  </main>
</div>
`;
